prov/tcp/src/tcpx_progress.c

```diff
@@ -434,9 +434,27 @@ void tcpx_progress(struct util_ep *util_ep)
 
 static int tcpx_try_func(void *util_ep)
 {
-	/* nothing to do here. When endpoints
-	   have incoming data, cq drives progress*/
+	uint32_t events;
+	struct util_wait_fd *wait_fd;
+	struct tcpx_ep *ep;
+
+	ep = container_of(util_ep, struct tcpx_ep, util_ep);
+	wait_fd = container_of(((struct util_ep *)util_ep)->rx_cq->wait,
+			       struct util_wait_fd, util_wait);
+
+	if (slist_empty(&ep->tx_queue) && !ep->send_ready_monitor) {
+		ep->send_ready_monitor = 1;
```

Comment:
I don't know if there is any coding style convention in Libfabric for this, but since `send_ready_monitor` is declared as a `boolean`, it should probably accept `true/false` instead of `1/0`.